### PR TITLE
Drop support for DAML-LF 1.3 from damlc

### PR DESCRIPTION
**Pull request by hurryabit**
This means `damlc` now only supports versions allowing arbitrary expressions
as contract keys. This is very helpful for the upcoming new template
de/re-sugaring underpinning generic templates.

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Add a line to the [release notes](https://github.com/digital-asset/daml/blob/master/docs/source/support/release-notes.rst), if appropriate

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

---
This change is [Reviewable](https://reviewable.io/reviews/digital-asset/daml/1445)

### CODE REVIEW

File: compiler/daml-lf-ast/src/DA/Daml/LF/Ast/Version.hs [hunk 1]
```diff
@@ -41,7 +37,7 @@ versionDev :: Version
 versionDev = V1 PointDev
 
 supportedInputVersions :: [Version]
-supportedInputVersions = [version1_3, version1_4, version1_5, versionDev]
+supportedInputVersions = [version1_4, version1_5, versionDev]
```
Comment (S11001001) [inline compiler/daml-lf-ast/src/DA/Daml/LF/Ast/Version.hs]: I think older versions (including 1.2) are still supported as _input_ with no changes, but it's the `supportedOutputVersions` below instead that is shrinking with these changes, right?

File: docs/source/support/release-notes.rst [hunk 2]
```diff
@@ -17,7 +17,7 @@ DAML Studio
 DAML Compiler
 ~~~~~~~~~~~~~
 
-- **BREAKING CHANGE**: Drop support for DAML-LF 1.2. Compiling to DAML-LF 1.3 should work without any code changes, although we highly recommend not specifying a target DAML-LF version at all.
+- **BREAKING CHANGE**: Drop support for DAML-LF 1.2 and 1.3. Compiling to DAML-LF 1.4 should work without any code changes, although we highly recommend not specifying a target DAML-LF version at all.
```
Comment (S11001001) [inline docs/source/support/release-notes.rst]: Dropping 1.2 has been released; this needs to be a separate note now.

File: daml-lf/tests/scenario/daml-1.3/contract-keys/EXPECTED.ledger [hunk 3]
```diff
@@ -1,8 +1,8 @@
 transactions:
```
Comment (bitonic) [inline daml-lf/tests/scenario/daml-1.3/contract-keys/EXPECTED.ledger]: these are not DAML 1.3 anymore, I'd just move them up a directory and use `dev`.

File: compiler/daml-lf-ast/src/DA/Daml/LF/Ast/Version.hs (hunk 1 above)
Comment (hurryabit) [reply compiler/daml-lf-ast/src/DA/Daml/LF/Ast/Version.hs]: Currently, all packages in an application need to be compiled using the same version of the compiler. Hence, there's no point in supporting versions for input that cannot be output.

File: daml-lf/tests/scenario/daml-1.3/contract-keys/EXPECTED.ledger (hunk 3 above)
Comment (hurryabit) [reply daml-lf/tests/scenario/daml-1.3/contract-keys/EXPECTED.ledger]: Done.

File: daml-lf/tests/scenario/actualize.sh [hunk 4]
```diff
@@ -18,11 +18,10 @@ TESTMAIN=$1
 TESTDIR="$(dirname $TESTMAIN)"
 TESTDAR="$TESTDIR/Main.dar"
 BAZEL_BIN="$(bazel info bazel-bin)"
-GHC_PRIM_DALF=${BAZEL_BIN}/daml-foundations/daml-ghc/package-database/deprecated/daml-prim-1.3.dalf
 REGEX_HIDE_HASHES="s,@[a-z0-9]{8},@XXXXXXXX,g"
 
 bazel build //daml-foundations/daml-tools/da-hs-damlc-app:da-hs-damlc-app
-../../../bazel-bin/daml-foundations/daml-tools/da-hs-damlc-app/da-hs-damlc-app package --debug --target 1.3 $TESTMAIN main -o $TESTDAR
+../../../bazel-bin/daml-foundations/daml-tools/da-hs-damlc-app/da-hs-damlc-app package --debug $TESTMAIN main -o $TESTDAR
```
Comment (bitonic) [inline daml-lf/tests/scenario/actualize.sh]: shouldn't this be `--target 1.dev`?

File: docs/source/support/release-notes.rst [hunk 5]
```diff
@@ -9,6 +9,11 @@ This page contains release notes for the SDK.
 HEAD — ongoing
 --------------
 
+DAML Compiler
+~~~~~~~~~~~~~
+
+- **BREAKING CHANGE**: Drop support for DAML-LF 1.3. Compiling to DAML-LF 1.4 should work without any code changes, although we highly recommend not specifying a target DAML-LF version at all.
```
Comment (bitonic) [inline docs/source/support/release-notes.rst]: I thought I had written this before, but please note that older packages _will still work with the new server_, it's just that you cannot compile with an older target.

File: daml-lf/tests/scenario/actualize.sh (hunk 4 above)
Comment (hurryabit) [reply daml-lf/tests/scenario/actualize.sh]: Good catch. Thanks.

File: docs/source/support/release-notes.rst (hunk 5 above)
Comment (hurryabit) [reply docs/source/support/release-notes.rst]: Addressed.

**Pull request merged**